theseus/src/api/profile/create.rs

```diff
@@ -154,6 +162,65 @@ pub async fn profile_create_from_creator(
     .await
 }
 
+pub async fn profile_create_from_duplicate(
+    copy_from: ProfilePathId,
+) -> crate::Result<ProfilePathId> {
+    let profile = profile::get(&copy_from, None).await?.ok_or_else(|| {
+        ErrorKind::UnmanagedProfileError(copy_from.to_string())
+    })?;
+
+    let profile_path_id = profile_create(
+        profile.metadata.name,
+        profile.metadata.game_version,
+        profile.metadata.loader,
+        profile.metadata.loader_version.map(|it| it.id),
+        profile.metadata.icon,
+        profile.metadata.icon_url,
+        profile.metadata.linked_data,
+        Some(true),
+        Some(true),
+    )
+    .await?;
+
+    // Copy it over using the import system (essentially importing from the same profile)
+    let state = State::get().await?;
+    let bar = pack::import::copy_dotminecraft(
+        profile_path_id.clone(),
+        copy_from.get_full_path().await?,
+        &state.io_semaphore,
+        None,
+    )
+    .await?;
+
+    if let Some(profile_val) =
+        crate::api::profile::get(&profile_path_id, None).await?
+    {
+        crate::launcher::install_minecraft(&profile_val, Some(bar)).await?;
```

Comment:
i don't think you need to run this function, since it will be the same exact profile

Reply:
Not sure why this was resolved- nothing changed